### PR TITLE
IGNITE-18491 Add tests for CLI completers

### PR DESCRIPTION
**Pull request by Flaugh24**
JdbcUrlDynamicCompleterFactory, ClusterUrlDynamicCompleterFactory, FilePathCompleter

https://issues.apache.org/jira/browse/IGNITE-18491

### CODE REVIEW

File: modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/core/repl/executor/ItIgnitePicocliCommandsTest.java [hunk 1]
```diff
@@ -356,6 +359,76 @@ void startStopNodeWhenCompleteNodeName() {
...
+        String[] expectedUrls = nodeNameRegistry.urls().stream().map(URL::toString).toArray(String[]::new);
+        assertThat("For given parsed words: " + parsedLine.words(),
+                complete(parsedLine),
+                containsInAnyOrder(expectedUrls));
+    }
+
+    @Test
+    @DisplayName("files suggested after -script-file option")
+    void suggestedScriptFile() {
+
```
Comment (PakhomovAlexander) [inline modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/core/repl/executor/ItIgnitePicocliCommandsTest.java:403]: looks like this empty line is not needed.

File: modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/core/repl/executor/ItIgnitePicocliCommandsTest.java [hunk 2]
```diff
@@ -356,6 +359,76 @@ void startStopNodeWhenCompleteNodeName() {
...
+        // Given
```
Comment (PakhomovAlexander) [inline modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/core/repl/executor/ItIgnitePicocliCommandsTest.java:404]: // Given files

File: modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/core/repl/executor/ItIgnitePicocliCommandsTest.java [hunk 3]
```diff
@@ -356,6 +359,76 @@ void startStopNodeWhenCompleteNodeName() {
...
+        String script1 = folder.getPath() + File.separator + "script1.sql";
+        Files.newFile(script1).deleteOnExit();
+
+        String script2 = folder.getPath() + File.separator + "script2.sql";
+        Files.newFile(script2).deleteOnExit();
+
+        String someFile = folder.getPath() + File.separator + "someFile.sql";
+        Files.newFile(someFile).deleteOnExit();
+
+        // Then
```
Comment (PakhomovAlexander) [inline modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/core/repl/executor/ItIgnitePicocliCommandsTest.java:419]: // When complete --script-file with folder typed

File: modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/core/repl/executor/ItIgnitePicocliCommandsTest.java [hunk 4]
```diff
@@ -356,6 +359,76 @@ void startStopNodeWhenCompleteNodeName() {
...
+
+        // Search in the folder
+        List<String> completions1 = complete(words("sql", "--script-file", folder.getPath()));
+        // Contains all files
+        assertThat(completions1, contains(script1, script2, someFile));
```
Comment (PakhomovAlexander) [inline modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/core/repl/executor/ItIgnitePicocliCommandsTest.java:424]: // Then completions contain all given files

File: modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/core/repl/executor/ItIgnitePicocliCommandsTest.java [hunk 5]
```diff
@@ -356,6 +359,76 @@ void startStopNodeWhenCompleteNodeName() {
...
+
+        // Search in the folder for files which starts with 'scrpit'
+        List<String> completions2 = complete(words("sql", "--script-file", folder.getPath() + File.separator + "script"));
```
Comment (PakhomovAlexander) [inline modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/core/repl/executor/ItIgnitePicocliCommandsTest.java:427]: // When complete --script-file with partial path to script

File: modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/core/repl/executor/ItIgnitePicocliCommandsTest.java [hunk 6]
```diff
@@ -356,6 +359,76 @@ void startStopNodeWhenCompleteNodeName() {
...
+        // Contains script1 and script2 files
+        assertThat(completions2, contains(script1, script2));
```
Comment (PakhomovAlexander) [inline modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/core/repl/executor/ItIgnitePicocliCommandsTest.java:429]: // Then ...

File: modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/core/repl/executor/ItIgnitePicocliCommandsTest.java [hunk 7]
```diff
@@ -356,6 +359,76 @@ void startStopNodeWhenCompleteNodeName() {
...
+
+        // Then
+        String[] expectedUrls = nodeNameRegistry.urls().stream().map(URL::toString).toArray(String[]::new);
+        assertThat("For given parsed words: " + parsedLine.words(),
+                complete(parsedLine),
+                containsInAnyOrder(expectedUrls));
+    }
+
+    @Test
+    @DisplayName("files suggested after -script-file option")
```
Comment (PakhomovAlexander) [inline modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/core/repl/executor/ItIgnitePicocliCommandsTest.java:401]: Could you please add the test case for the folder that contains only folders but not files? Would it be completed correctly?

**Review by PakhomovAlexander — CHANGES_REQUESTED** [review]

File: modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/core/repl/executor/ItIgnitePicocliCommandsTest.java (hunk 2 above)
Comment (Flaugh24) [reply modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/core/repl/executor/ItIgnitePicocliCommandsTest.java]: done

File: modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/core/repl/executor/ItIgnitePicocliCommandsTest.java (hunk 7 above)
Comment (Flaugh24) [reply modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/core/repl/executor/ItIgnitePicocliCommandsTest.java:401]: done

File: modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/core/repl/executor/ItIgnitePicocliCommandsTest.java (hunk 1 above)
Comment (Flaugh24) [reply modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/core/repl/executor/ItIgnitePicocliCommandsTest.java:404]: done

File: modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/core/repl/executor/ItIgnitePicocliCommandsTest.java (hunk 3 above)
Comment (Flaugh24) [reply modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/core/repl/executor/ItIgnitePicocliCommandsTest.java]: done

File: modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/core/repl/executor/ItIgnitePicocliCommandsTest.java (hunk 4 above)
Comment (Flaugh24) [reply modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/core/repl/executor/ItIgnitePicocliCommandsTest.java]: done

**Review by PakhomovAlexander — APPROVED** [review]

**Pull request merged**